### ISSUE TITLE
Accessibility warnings that rely on event directives are wrong

### ISSUE THREAD

**Issue by cdcarson** [sveltejs/language-tools#2095]
### Describe the bug

This perfectly legit code...
```svelte
<div on:click={handler}><slot/></div>
```

...produces these a11y errors...

```
A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event. svelte(a11y-click-events-have-key-events)
A11y: <div> with click handler must have an ARIA role svelte(a11y-no-static-element-interactions)
```
This is wrong in two ways:

- Events propagate. Simply attaching an event to a non-interactive element has no effect _in and of itself_ on a11y or the semantics on which AT relies. (If it does -- if user agents do indeed infer the role of an element from the event listeners that have been attached to it -- then I'm wrong. But that would be...ugh...their problem?)
- More importantly. The a11y errors and suggestions Svelte displays encourage, rather than discourage, bad a11y practices. "Sure buddy, it's ok to make your div a link. Just mess around until you hit on the right combination." This can easily lead to the following...

```svelte
<div on:click={handler} role="button" tabindex="0" on:keyup={() => {}}>
  <slot/>
</div>
```
...which passes Svelte's a11y linting with flying colors even though it is obviously tripe. Less obviously, if my intention was  simply to have the `div` listen for clicks, rather than turn it into a button, then the linter, rather than helping, has done significant harm, both to the accessibility of the end product and to my (the developer's) understanding of a11y and javascript.

Fix: A11y linting rules that rely on Svelte event directives (e.g. `on:eventName`) should be gotten rid of. There's no way (at least that I can think of) to divine what the presence of a listener implies. 

There's an [issue on this repo](https://github.com/sveltejs/language-tools/issues/650) where folks have been talking about ways to disable a11y warnings in VsCode, etc. It's been open for two and a half years. Wouldn't it be better just to fix the the underlying problems rather than continue to tell people that they can skirt around the defects by doing this, that and the other?

### Reproduction

see above

### Expected behaviour

see above

### System Info

n/a

### Which package is the issue about?

svelte-language-server

**Comment by jasonlyu123** [comment]
It's indeed a false positive for catching bubbled event. But the compiler won't know that there is a button inside the slot. That's why there is the `<!--svelte-ignore-->` comment that can be used to disable when you're sure it is a false positive. 
https://svelte.dev/docs/accessibility-warnings

It's way more common for people to use `on:event` on the exact element the event triggers. We're not going to give up most of the cases we can catch because of the false positives. 

As for hints about using a button instead, there is already an issue in the svelte core https://github.com/sveltejs/svelte/issues/8001.

This is overall a svelte core issue, so I'll close this later in favour of the upstream issue. What we can do is add a code action quick fix to turn div into a button. And maybe even add styles to remove the border and background. But I'll prefer to add this after svelte core add the hint to the warning message.

**Comment by cdcarson** [comment]
> It's way more common for people to use on:event on the exact element the event triggers. We're not going to give up most of the cases we can catch because of the false positives.

I disagree, The salient case is listening for child events, which it would be inconvenient or impossible to attach the relevant listeners.

Leave this open, please.
